packages/shared_preferences/shared_preferences_web/lib/shared_preferences_web.dart

```diff
@@ -31,9 +31,9 @@ class SharedPreferencesPlugin extends SharedPreferencesStorePlatform {
 
   @override
   Future<Map<String, Object>> getAll() async {
-    final Map<String, Object> allData = <String, Object>{};
+    final Map<String, Object> allData = {};
     for (String key in _storedFlutterKeys) {
-      allData[key] = _decodeValue(html.window.localStorage[key]);
+      allData[key] = _decodeValue(html.window.localStorage[key]!);
```

Comment:
I'm not familiar with the web implementation; can you explain why this is safe?

Reply:
We iterate through all keys so we are sure that every key has data. I just realized that I forgot the case where the case where the data is actually null. Will fix it now.

Reply:
Edit : localStorage always returns a string and is then parsed back to an `Object` using `jsonDecode`. Thus even if the data is null its value is actually `"null"`. The only case where it's actually `null` is when we don't have the right key which can't be the case as we iterate through all the keys with `_storedFlutterKeys`.

![image](https://user-images.githubusercontent.com/25441359/107127011-9b5e1100-68b3-11eb-8d5f-5f92263f992d.png)